src/components/PostDetails/PostDetails.jsx

```diff
@@ -1,45 +1,115 @@
-import React from 'react';
-import { NewCommentForm } from '../NewCommentForm';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
 import './PostDetails.scss';
+import { Loader } from '../Loader';
+import { NewCommentForm } from '../NewCommentForm';
+import { getPostDetails } from '../../api/posts';
+import * as api from '../../api/comments';
+
+export const PostDetails = ({ postId }) => {
+  const [post, setPost] = useState(null);
+  const [comments, setComments] = useState(null);
+  const [hiddenComments, setHiddenComments] = useState(false);
+
+  useEffect(() => {
+    const fetchAllData = async() => {
+      const postInfo = await getPostDetails(postId);
+      const commentsInfo = await api.getPostComments(postId);
+
+      setPost(postInfo.data);
+      setComments(commentsInfo.data);
+    };
+
+    fetchAllData();
+  }, [postId]);
+
+  const deleteComment = async(commentId) => {
+    await api.deletePostComment(commentId);
+    const newComments = await api.getPostComments(postId);
+
+    setComments(newComments.data);
+  };
+
+  const addComment = async({ name, email, body }) => {
+    await api.addPostComment({
+      postId, name, email, body,
+    });
+    const newComments = await api.getPostComments(postId);
+
+    setComments(newComments.data);
+  };
+
+  if (postId && !post) {
+    return <Loader />;
+  }
+
+  return (
+    <>
+      {post
+        && (
+          <div className="PostDetails">
+            <h2>Post details:</h2>
+
+            <section className="PostDetails__post">
+              <p>{post.body}</p>
+            </section>
+
+            {comments && comments.length > 0
```

Comment:
```suggestion
            {comments.length > 0
```
It should be enough

Reply:
Thanks for the suggestion, this one will really help cut the code)